crates/sui-move-build/src/lib.rs

```diff
@@ -235,8 +240,14 @@ pub fn build_from_resolution_graph(
     run_bytecode_verifier: bool,
     print_diags_to_stderr: bool,
     chain_id: Option<String>,
+    with_unpublished_dependencies: bool,
 ) -> SuiResult<CompiledPackage> {
-    let (published_at, dependency_ids) = gather_published_ids(&resolution_graph, chain_id);
+    let (published_at, mut dependency_ids) = gather_published_ids(&resolution_graph, chain_id);
+
+    check_invalid_dependencies(&dependency_ids.invalid)?;
+    if !with_unpublished_dependencies {
+        check_unpublished_dependencies(&dependency_ids.unpublished)?;
+    };
```

Comment:
This was moved from `compile_package` in `sui/src/client_commands.rs` to here as it makes more logical sense to keep as many checks as possible related to dependencies in this function.